src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/WithRepresentativePersonalisationTest.java

```diff
@@ -32,69 +31,96 @@
+    private String hearingPostponedRepsEmailId;
+
     public WithRepresentativePersonalisationTest() {
         super(30);
     }
 
     @Test
     @Parameters(method = "eventTypeAndSubscriptions")
-    public void givenEventAndRepsSubscription_shouldSendNotificationToReps(NotificationEventType notificationEventType)
-            throws Exception {
+    public void givenEventAndRepsSubscription_shouldSendNotificationToReps(NotificationEventType notificationEventType,
+                                                                           boolean isSmsNotificationFound)
+        throws Exception {
         //Given
         final String appellantEmailId = getFieldValue(notificationEventType, "AppellantEmailId");
         final String appellantSmsId = getFieldValue(notificationEventType, "AppellantSmsId");
         final String repsEmailId = getFieldValue(notificationEventType, "RepsEmailId");
         final String repsSmsId = getFieldValue(notificationEventType, "RepsSmsId");
 
         simulateCcdCallback(notificationEventType,
-                "representative/" + notificationEventType.getId() + "Callback.json");
-
-        List<Notification> notifications = tryFetchNotificationsForTestCase(
-            appellantEmailId, appellantSmsId,
-            repsEmailId, repsSmsId);
-
-        assertNotificationBodyContains(notifications, appellantEmailId);
-        assertNotificationBodyContains(notifications, appellantSmsId);
+            "representative/" + notificationEventType.getId() + "Callback.json");
+
+        List<Notification> notifications;
+        if (isSmsNotificationFound) {
+            notifications = tryFetchNotificationsForTestCase(
+                appellantEmailId, appellantSmsId, repsEmailId, repsSmsId);
+            assertNotificationBodyContains(notifications, appellantEmailId);
+            assertNotificationBodyContains(notifications, appellantSmsId);
+        } else {
+            notifications = tryFetchNotificationsForTestCase(
+                appellantEmailId, repsEmailId);
+            assertNotificationBodyContains(notifications, appellantEmailId);
+        }
 
         String representativeName = "Harry Potter";
         assertNotificationBodyContains(notifications, repsEmailId, representativeName);
-        assertNotificationBodyContains(notifications, repsSmsId);
+
+        if (isSmsNotificationFound) {
+            assertNotificationBodyContains(notifications, repsSmsId);
+        }
     }
 
     @Test
     @Parameters(method = "eventTypeAndSubscriptions")
-    public void givenEventAndNoRepsSubscription_shouldNotSendNotificationToReps(NotificationEventType notificationEventType)
-            throws Exception {
+    public void givenEventAndNoRepsSubscription_shouldNotSendNotificationToReps(NotificationEventType notificationEventType,
+                                                                                boolean isSmsNotificationFound)
```

Comment:
I'd probably write a email only test rather than retrofit a notification type which only sends emails and not sms messages